### PR TITLE
✨ Add BuildConfig to provenance

### PR DESCRIPTION
**Pull request by laurentsimon**
see discussion in https://github.com/gossts/slsa-go/issues/10

### CODE REVIEW

File: pkg/provenance.go [hunk 1]
```diff
@@ -100,11 +117,22 @@ func GenerateProvenance(name, digest, githubContext string) ([]byte, error) {
...
-			Materials: []slsa.ProvenanceMaterial{{
-				URI: fmt.Sprintf("git+%s.git", gh.Repository),
-				Digest: slsa.DigestSet{
-					"SHA1": gh.SHA,
-				}},
+			BuildConfig: BuildConfig{
+				Steps: []Step{
+					// Single step.
+					{
+						Command: com,
```
Comment (asraa) [inline pkg/provenance.go:124]: I think the path to the go compiler also be listed here, maybe in the Command. I don't think `com` includes the `/path/to/golang`. WDYT?

Comment (laurentsimon) [reply pkg/provenance.go:124]: you mean it should contain the compiler's path as well, correct? If so, it is already included, see https://github.com/laurentsimon/slsa-go/blob/feat/buildconfig/pkg/build.go#L96

**Review by asraa — APPROVED** [review]

File: pkg/provenance.go (hunk 1 above)
Comment (Dreamypops33) [reply pkg/provenance.go:124]: Sorry for not responding to your text and emails. My family was I'll my dad has had several strokes and suffers from dementia. So i have not been as loyal them the COVID had a very strong impact on myself ending up homeless, barely able to eat considering i have a child. I was taking care of such a wonderful role model dad, a overworked retired mom and almost lost all time to focus on this wonderful development. Perhaps it's not to late to continue this great development. JASON HOWARD IS MY NE.. EMAIL ME jhowardstop@outlook.com OR JUST CALL 5014955893.  PLEASE DONT BE Shy.

**Comment by Dreamypops33** [comment]
Sorry for not responding to your text and emails. My family was I'll my dad has had several strokes and suffers from dementia. So i have not been as loyal them the COVID had a very strong impact on myself ending up homeless, barely able to eat considering i have a child. I was taking care of such a wonderful role model dad, a overworked retired mom and almost lost all time to focus on this wonderful development. Perhaps it's not to late to continue this great development. JASON HOWARD IS MY NE.. EMAIL ME jhowardstop@outlook.com OR JUST CALL 5014955893.  PLEASE DONT BE Shy.

**Pull request merged**